.github/workflows/ci.yml

```diff
@@ -0,0 +1,73 @@
+
+jobs:
+  unit-test:
+    name: Rustfmt, Clippy and Unit Test
+    if: |
+      github.event.pull_request.draft == false &&
+      (github.event.action == 'ready_for_review' || needs.skip_check.outputs.should_skip != 'true')
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+      - name: Cargo cache
+        uses: Swatinem/rust-cache@v2
+      - name: Cargo check
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --all-features
+      - name: Rustfmt check
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+      - name: Run tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --verbose --release --all
+
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+      - name: Cargo cache
+        uses: Swatinem/rust-cache@v2
+      - name: Build integration-test
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --bin integration-test --release
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '~1.18'
+      - uses: actions/setup-node@v3
+        node-version: 18
+        cache: 'npm'
+      - name: Build geth
+        run: go install github.com/scroll-tech/go-ethereum/cmd/geth@develop
+      - name: Init geth
+        run: |
+          geth --datadir $GITHUB_WORKSPACE/geth/ init ./integration-tests/genesis.json
+          cp ./integration-tests/keystore $GITHUB_WORKSPACE/geth/keystore/
+          echo -n scroll > password
+      - name: Run Tests
+        working-directory: ./integration-tests
+        run: |
+          geth --datadir=$GITHUB_WORKSPACE/geth/ --unlock 0x4cb1ab63af5d8931ce09673ebd8ae2ce16fd6571 --password $GITHUB_WORKSPACE/password --mine --allow-insecure-unlock --http --http.api=net,eth,scroll --maxpeers 0 --trace.mptwitness=1 &
```

Comment:
And here also use `--trace.mptwitness=2`

However, considering we woud deprecated the witness generation later (see [#PR257 in l2geth](https://github.com/scroll-tech/go-ethereum/pull/257)), I am not sure how long this CI can keep working in mpt repo. Maybe we can move it into scroll-zkevm (we make mpt witness in zkevm-circuit now, and unit test should be enough for a crate-only project)